pkg/orchestrator/assetscanestimationwatcher/watcher.go

```diff
@@ -0,0 +1,269 @@
+		if !ok {
+			logger.Warnf("Skipping due to invalid AssetScanEstimation: Asset.ID is nil: %v", assetScanEstimation)
+			continue
+		}
+
+		events = append(events, AssetScanEstimationReconcileEvent{
+			AssetScanEstimationID: assetScanEstimationID,
+			ScanEstimationID:      scanEstimationID,
+			AssetID:               assetID,
+		})
+	}
+
+	return events, nil
+}
+
+// nolint:cyclop
+func (w *Watcher) Reconcile(ctx context.Context, event AssetScanEstimationReconcileEvent) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx).WithFields(event.ToFields())
+	ctx = log.SetLoggerForContext(ctx, logger)
+
+	assetScanEstimation, err := w.backend.GetAssetScanEstimation(ctx, event.AssetScanEstimationID, models.GetAssetScanEstimationsAssetScanEstimationIDParams{
+		Expand: utils.PointerTo("asset"),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get AssetScanEstimation with %s id: %w", event.AssetScanEstimationID, err)
+	}
+
+	state, ok := assetScanEstimation.GetState()
+	if !ok {
+		return fmt.Errorf("cannot determine state of AssetScanEstimation with %s id", event.AssetScanEstimationID)
+	}
+
+	logger.Tracef("Reconciling AssetScanEstimation state: %s", state)
+
+	switch state {
+	case models.AssetScanEstimationStateStatePending:
+		if err = w.reconcilePending(ctx, &assetScanEstimation); err != nil {
+			return err
+		}
+	case models.AssetScanEstimationStateStateAborted:
+		if err = w.reconcileAborted(ctx, &assetScanEstimation); err != nil {
+			return err
+		}
+	default:
+	}
+
+	return nil
+}
+
+// nolint:cyclop
+func (w *Watcher) reconcilePending(ctx context.Context, assetScanEstimation *models.AssetScanEstimation) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+
+	logger.Debugf("reconciling pending asset scan estimations")
+
+	assetScanEstimationID, ok := assetScanEstimation.GetID()
+	if !ok {
+		return errors.New("invalid AssetScanEstimation: ID is nil")
+	}
+
+	if assetScanEstimation.Asset == nil || assetScanEstimation.Asset.AssetInfo == nil {
+		return errors.New("invalid AssetScanEstimation: Asset or AssetInfo is nil")
+	}
+
+	asset := &models.Asset{
+		AssetInfo:    assetScanEstimation.Asset.AssetInfo,
+		FirstSeen:    assetScanEstimation.Asset.FirstSeen,
+		Id:           &assetScanEstimation.Asset.Id,
+		LastSeen:     assetScanEstimation.Asset.LastSeen,
+		Revision:     assetScanEstimation.Asset.Revision,
+		ScansCount:   assetScanEstimation.Asset.ScansCount,
+		Summary:      assetScanEstimation.Asset.Summary,
+		TerminatedOn: assetScanEstimation.Asset.TerminatedOn,
+	}
+
+	stats := w.getLatestAssetScanStats(ctx, asset)
+
+	estimation, err := w.provider.Estimate(ctx, stats, asset, assetScanEstimation.AssetScanTemplate)
+	if err != nil {
+		logger.Errorf("Failed to estimate asset scan: %v", err)
+		assetScanEstimation.State.State = utils.PointerTo(models.AssetScanEstimationStateStateFailed)
+		assetScanEstimation.State.Errors = utils.PointerTo(utils.UnwrapErrorStrings(err))
```

Comment:
replace with:
```
assetScanEstimation.State.Message = utils.PointerTo(err.Error())
assetScanEstimation.State.Reason = utils.PointerTo(models.AssetScanEstimationStateReasonUnexpected)
```

Reply:
We need to check for Fatal errors here, we should only be moving directly to failed for Fatal errors, any other error should be bubbled up to the Reconciler to handle Retryable/unexpected errors which can be retried. Timeout should handle any case when we get stuck in a retry loop.